### DISCUSSION TITLE
Payrun is being aborted

### DISCUSSION THREAD

**Discussion by Zahab-Ifisol** [Payroll-Engine/PayrollEngine#3] [Q&A]
We have followed the procedure described in the Basic Payroll Example. We executed the process step by step. However, whenever we execute a payrun, the status of the payrun comes as aborted. For context , we have 2 employees currently under the same division and tenant. We also created employee cases for these two employees as described in the documentation. To keep things simple, we just created a salary case for now. 
Upon studying the response of the payrun job API, we came to know that the employee count and processed employee count is coming as 0 for some reason. Screenshot is attached. Please advise and guide 
![image_2024_03_15T10_33_43_768Z](https://github.com/Payroll-Engine/PayrollEngine/assets/103321860/20725108-1854-460b-a1ac-38d074bdc9d2)

**Comment by Giannoudis** [comment]
Can you give me your example as a json?

**Comment by Zahab-Ifisol** [comment]
Giannoudis thankyou for reaching out! The following is our JSON of the Regulation, which we believe is causing the issue. Does help you figure out the problem?
What other JSONs do you need from us 

{
    "name": "Payroll",
    "nameLocalizations": {
      "additionalProp1": "string",
      "additionalProp2": "string",
      "additionalProp3": "string"
    },
    "version": 1,
    "sharedRegulation": true,
    "validFrom": "2024-03-01T07:26:44.383",
    "owner": "Ifisol",
    "description": null,
    "descriptionLocalizations": {
      "additionalProp1": "string",
      "additionalProp2": "string",
      "additionalProp3": "string"
    },
    "baseRegulations": null,
    "attributes": null,
    "id": 16,
    "status": "Active",
    "created": "2024-03-01T07:26:44.383",
    "updated": "2024-03-18T08:44:59.096099"
  },

**Comment by Giannoudis** [comment]
Thanks for the example. However, this is only the payroll part. Do you also have the jsons for the employee and the regulations?

**Comment by Zahab-Ifisol** [comment]
yes please , I am going to attach the requested items 

Employees 


 {
    "identifier": "Imran",
    "firstName": "Imran",
    "lastName": "Zahab",
    "divisions": [
      "Company1"
    ],
    "culture": null,
    "calendar": null,
    "attributes": null,
    "id": 15,
    "status": "Active",
    "created": "2024-02-27T11:10:03.0877579",
    "updated": "2024-02-27T11:10:03.0877579"
  },
  {
    "identifier": "Carmel",
    "firstName": "Carmel",
    "lastName": "Test",
    "divisions": [
      "Company1"
    ],
    "culture": null,
    "calendar": null,
    "attributes": null,
    "id": 19,
    "status": "Active",
    "created": "2024-03-11T10:39:34.1618331",
    "updated": "2024-03-11T10:39:34.1618331"
  }

**Comment by Zahab-Ifisol** [comment]
Cases 

  "caseType": "Employee",
    "name": "Salary",
    "nameLocalizations": null,
    "nameSynonyms": null,
    "description": null,
    "descriptionLocalizations": null,
    "defaultReason": null,
    "defaultReasonLocalizations": null,
    "baseCase": null,
    "baseCaseFields": null,
    "overrideType": "Active",
    "cancellationType": "None",
    "hidden": false,
    "availableExpression": null,
    "buildExpression": "true",
    "validateExpression": "true",
    "lookups": null,
    "slots": null,
    "availableActions": null,
    "buildActions": null,
    "validateActions": null,
    "attributes": null,
    "clusters": null,
    "id": 86,
    "status": "Active",
    "created": "2023-01-01T00:00:00",
    "updated": "2023-01-01T00:00:00"
  }

**Comment by Zahab-Ifisol** [comment]
Wage type

{
    "wageTypeNumber": 100,
    "name": "Salary",
    "nameLocalizations": null,
    "description": "test",
    "descriptionLocalizations": null,
    "overrideType": "Active",
    "valueType": "Money",
    "calendar": null,
    "collectors": [
      "Income"
    ],
    "collectorGroups": null,
    "valueExpression": "var values = GetCaseValues(\"Salary\"); return values[\"Salary\"];",
    "resultExpression": null,
    "attributes": null,
    "clusters": null,
    "id": 40,
    "status": "Active",
    "created": "2024-03-01T11:19:31.465",
    "updated": "2024-03-01T12:41:56.7674695"
  }

**Comment by Zahab-Ifisol** [comment]
{
    "name": "Income",
    "nameLocalizations": null,
    "collectMode": "Summary",
    "negated": false,
    "overrideType": "Active",
    "valueType": "Money",
    "collectorGroups": null,
    "threshold": null,
    "minResult": null,
    "maxResult": null,
    "startExpression": null,
    "applyExpression": null,
    "endExpression": null,
    "attributes": null,
    "clusters": null,
    "id": 17,
    "status": "Active",
    "created": "2024-03-01T07:49:03.9331738",
    "updated": "2024-03-01T07:49:03.9331738"
  },
  {
    "name": "Deduction",
    "nameLocalizations": null,
    "collectMode": "Summary",
    "negated": true,
    "overrideType": "Active",
    "valueType": "Money",
    "collectorGroups": null,
    "threshold": null,
    "minResult": null,
    "maxResult": null,
    "startExpression": null,
    "applyExpression": null,
    "endExpression": null,
    "attributes": null,
    "clusters": null,
    "id": 18,
    "status": "Active",
    "created": "2024-03-01T07:49:28.6345254",
    "updated": "2024-03-01T07:49:28.6345254"
  }
  
  collectors

**Comment by Zahab-Ifisol** [comment]
Let me know if there is anything else that you need

**Comment by Giannoudis** [comment]
Thanks for the sample data. There are several issues:
- The payroll needs at least one payroll layer
- The payroll needs a division `Company1`.
- The case requires case fields
- The wage type `ValueExpression` is invalid: try `CaseValue["Salary"]`.

Here is your example with test [Test.pt.json](https://github.com/Payroll-Engine/PayrollEngine/files/14637497/Test.pt.json)
See also the [Basic Example](https://github.com/Payroll-Engine/PayrollEngine/blob/main/Examples/StartPayroll/Basic.json).

**Reply by Zahab-Ifisol** [reply]
Thankyou ! we will try and incorporate these changes and let you know !

**Comment by Zahab-Ifisol** [comment]
Would you be able to guide how we can set up the payroll console ? Will the payroll console allow us to execute the JSON you shared with us in a single step ?
Cause currently we are using swagger and it gets pretty complicated to use

**Comment by Giannoudis** [comment]
The REST API is used by client applications for the development of payroll solutions. There are two types of clients:

**Console Client**
JSON can be used to develop and automatically test payroll solutions. This requires appropriate knowledge of writing JSON files.
Payroll Console Setup: https://github.com/Payroll-Engine/PayrollEngine/wiki/Setup
Payroll Console usage: https://github.com/Payroll-Engine/PayrollEngine.PayrollConsole

**Web Client**
With the Web Client, payroll development takes place in a user-friendly application. This is probably the best option for beginners.
Payroll Web Client Setup: https://github.com/Payroll-Engine/PayrollEngine/wiki/Setup
Payroll Web Client configuration: https://github.com/Payroll-Engine/PayrollEngine.WebApp

The following documents are helpful:
- Example Basic Payroll (Console and Web) - https://github.com/Payroll-Engine/PayrollEngine/wiki/Basic-Payroll
- Introduction Payroll Engine - https://dev.to/giannoudis/introducing-payroll-engine-114o
- Payroll Engine Whitepaper - https://github.com/Payroll-Engine/PayrollEngine/blob/main/docs/PayrollEngineWhitepaper.pdf
